src/mimefactory.rs

```diff
@@ -1111,29 +1144,33 @@ impl MimeFactory {
                         for (addr, key) in &encryption_pubkeys {
                             let fingerprint = key.dc_fingerprint().hex();
                             let cmd = msg.param.get_cmd();
-                            let should_do_gossip = cmd == SystemMessage::MemberAddedToGroup
-                                || cmd == SystemMessage::SecurejoinMessage
-                                || multiple_recipients && {
-                                    let gossiped_timestamp: Option<i64> = context
-                                        .sql
-                                        .query_get_value(
-                                            "SELECT timestamp
+                            let is_full_msg =
+                                self.pre_message_mode == Some(PreMessageMode::FullMessage);
+                            let should_do_gossip = !is_full_msg
```

Comment:
Better do
```Rust
if !is_full_msg {
    continue;
}
```
then remormatting the code isn't needed, and the `assert!` below too which looks very protective (usually we don't assert just checked conditions)